### PR TITLE
[sdk] Add support for access token authentication

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Adds basic support for `user: { accessToken: '...' }` to programmatically interact with stored Snacks.

# How

- Added helper for the `Expo-Session` or `Authorization: bearer ...` headers.
- Added tests for devsession to send authenticated requests

# Test Plan

- Create a new snack
- Add `snack.setUser({ accessToken: '...' })`
- Save the snack
- Snack should be saved on account by only using the access token

### CODE REVIEW

File: packages/snack-sdk/src/DevSession.ts [hunk 1]
```diff
@@ -66,8 +66,8 @@ export default class DevSession {
       url,
       method: 'post',
       headers: {
-        ...(user?.sessionSecret ? { 'Expo-Session': user.sessionSecret } : {}),
         'Content-Type': 'application/json',
+        ...createUserHeader(user),
```
Comment (IjzerenHein) [inline packages/snack-sdk/src/DevSession.ts:70]: Nice!

File: packages/snack-sdk/src/utils.ts [hunk 2]
```diff
@@ -62,3 +62,15 @@ export function createError(config: {
...
   return error;
 }
+
+export function createUserHeader(user?: SnackUser): { [key: string]: string } {
+  if (user?.sessionSecret) {
+    return { 'Expo-Session': user.sessionSecret };
+  }
+
+  if (user?.accessToken) {
+    return { Authorization: `bearer ${user.accessToken}` };
```
Comment (IjzerenHein) [inline packages/snack-sdk/src/utils.ts:72]: Does it make any difference how `Bearer` is capitalized?

```suggestion
    return { Authorization: `Bearer ${user.accessToken}` };
```

**Review by IjzerenHein — CHANGES_REQUESTED** [review]
Looks good 👍 

We should also add tests for Saving, which is probably the most important scenario to verify (much more than dev-sessions).

File: packages/snack-sdk/src/utils.ts (hunk 2 above)
Comment (byCedric) [reply packages/snack-sdk/src/utils.ts:72]: You are right, it should be capitalized, afaik bearer tokens originates from OAuth. [In that spec](https://tools.ietf.org/html/rfc6750#section-2.1) it's defined with a capital. We also [use the capitalized one in CLI](https://github.com/expo/expo-cli/blob/master/packages/xdl/src/ApiV2.ts#L205)

**Comment by IjzerenHein** [comment]
friendly ping to @byCedric :)

**Comment by byCedric** [comment]
@IjzerenHein done, added the same test case for `accessToken` and `sessionSecret` for snack saves.

**Review by IjzerenHein — APPROVED** [review]

**Comment by IjzerenHein** [comment]
Awesome! Noticed that the tests weren't completing. This is because each snack that is `online`, also needs to be turned `offline`.

**Pull request merged**